### DISCUSSION TITLE
ot-efr32 RSSI issues

### DISCUSSION THREAD

**Discussion by kmfnjtze8888** [openthread/openthread#7750] [Q&A] [answered]
Hello,

I created an ot-cli-ftd binary from the **ot-efr32** repo, but I seem to be getting issues with the `RSSI` values.

I am using brd4170a radio board with an efr32mg12 chip. I wanted to use `915 MHz` with `OQPSK` modulation, so I changed the macros in ot-efr32/src/efr32mg12/brd4170a/board_config.h

![Picture1](https://user-images.githubusercontent.com/72487638/170336804-98a467b9-218e-4d4f-8f19-6f19974693a2.jpg)

I built the project, then flashed ot-cli-ftd to two brd4170a boards. However, when I try to scan for the RSSI between them, it seems 50 dBm lower than I expected. It says the `RSSI` is -87 and -88, when it should be -37, -38.

![image](https://user-images.githubusercontent.com/72487638/170338776-599a3b78-93c4-4f11-b84a-9d86ccbd2fd8.png)

![image](https://user-images.githubusercontent.com/72487638/170338837-7d5af8dd-17fb-4efa-835a-06264cf8f85b.png)

Can someone point me to the right direction on how to debug this?

**Comment by jwhui** [comment]
@schinchani may be able to provide some insight.

Also note that the sub-GHz feature is experimental and not part of the Thread Specification.

**Reply by schinchani** [reply]
@kmfnjtze8888 - I am currently investigating this at my end and will get back to you as soon as I have more info. Thank you.

**Reply by kmfnjtze8888** [reply]
Hi @schinchani, any updates?

**Comment by kmfnjtze8888** [comment]
Hello @schinchani, just wondering if you guys have any more info regarding this at the moment? Sorry if I already asked twice, as this is quite critical for us. Thank you

**Comment by schinchani** [comment]
Hello @kmfnjtze8888 

Thank you for your patience and sincere apologies for the delay here.  Quick question - are you positive that you are using the right antenna when operating on the 915 MHz band? Not using an antenna or an incorrect match will impact the RSSI values.

As a next step, I am trying to reproduce the issue at my end and also investigating this with the platform team at our end. I will keep you posted as soon as I have any further updates in this regard.

Thanks.

**Reply by kmfnjtze8888** [reply]
Hi @schinchani,

Yup, I am certain that I'm using the correct antenna because I tested my old radio firmware (Thread 1.1) with the same hardware setup and I was receiving the correct RSSI values on the 915 MHz band. I also tried connecting two radio boards directly with an RF cable with 40 dB attenuators, and I was still getting incorrect RSSI values.

Thank you for the update!

**Comment by schinchani** [comment, marked as answer]
Hi @kmfnjtze8888 

We suspect this to be an RSSI offset corruption issue. Can you try replacing lines 161 and 162 of https://github.com/openthread/ot-efr32/blob/main/src/efr32mg12/phy/PHY_IEEE802154_915MHZ_OQPSK_EFR32XG12.c#L161

`   
0x00077014UL,
0x000270FEUL,
`

with the following values and let me know if this fixes the issue for you:

`   
0x10017014UL, 0x0007F800UL,
0x30017014UL, 0x000000FEUL,
0x00067018UL,
`

Please note that the sub-GHz PHY support is experimental, not a part of the Thread specification and has not been exhaustively tested.

Thanks
Sagar

**Reply by kmfnjtze8888** [reply]
Hi @schinchani ,

Looks like that fixed the RSSI issues! We will be doing some more tests to verify that no other issue arises. Thank you so much! You're awesome!

**Reply by schinchani** [reply]
Thank you for letting us know. Good luck.

Regards,
Sagar.

**Reply by kmfnjtze8888** [reply]
Hi @schinchani,

Just a quick question, are positive RSSI values valid? Because I thought RSSI values would always be negative? Thanks
